### ISSUE TITLE
Optimize readUntilPrompt function

### ISSUE THREAD

**Issue by karimra** [networklore/netrasp#16]
Hi,

Since I spent some time looking at the readUntilPrompt function for #11 and #12 , I ended up trying to rewrite it to be more compliant with Go best practices. 
Instead of using a `string` to build the output, the two other options are `strings.Builder` and `bytes.Buffer`. 

I tried both approaches and benchmarked the 3 implementations, the original one, using `strings.Builder` and `bytes.Buffer`. 
The dummy terminal outputs a variable number of lines, 50 bytes in each line except the last one which is `prompt#`

```
goos: linux
goarch: amd64
pkg: github.com/networklore/netrasp/pkg/netrasp
BenchmarkReaderOriginal100Lines            91527             13447 ns/op           51681 B/op         20 allocs/op
BenchmarkReaderStringsBuilder100Lines      88066             13549 ns/op           37649 B/op         16 allocs/op
BenchmarkReaderBytesBuffer100Lines        112674             10637 ns/op           42993 B/op         16 allocs/op

BenchmarkReaderOriginal1KLines              2917            420302 ns/op         1412228 B/op         85 allocs/op
BenchmarkReaderStringsBuilder1KLines        2725            411526 ns/op          770236 B/op         65 allocs/op
BenchmarkReaderBytesBuffer1KLines           4234            283635 ns/op          774140 B/op         58 allocs/op

BenchmarkReaderOriginal10KLines               40          29583267 ns/op        82506021 B/op        761 allocs/op
BenchmarkReaderStringsBuilder10KLines         42          28545821 ns/op        33242154 B/op        520 allocs/op
BenchmarkReaderBytesBuffer10KLines            68          17569077 ns/op        31843859 B/op        461 allocs/op

BenchmarkReaderOriginal50KLines                2         688776821 ns/op        1966611136 B/op     3954 allocs/op
BenchmarkReaderStringsBuilder50KLines          2         653479936 ns/op        723106100 B/op      2550 allocs/op
BenchmarkReaderBytesBuffer50KLines             3         411061399 ns/op        714666413 B/op      2279 allocs/op

BenchmarkReaderOriginal100KLines               1        2952538159 ns/op        7817518688 B/op     8233 allocs/op
BenchmarkReaderStringsBuilder100KLines         1        2578991040 ns/op        2837066472 B/op     5110 allocs/op
BenchmarkReaderBytesBuffer100KLines            1        1784562293 ns/op        2807405400 B/op     4546 allocs/op
PASS
ok      github.com/networklore/netrasp/pkg/netrasp      25.120s
```

As you can see the `bytes.Buffer` implementation is 20 to 40% more efficient than the `string` one.

So I was wondering if you would be interested in a PR that replaces `readUntilPrompt` implementation ?

**Comment by ogenstad** [comment]
Hi,

That sounds good, I'd be happy to include any improvements like this!

**Comment by ogenstad** [comment]
Closing as this was fixed in #17.
